pages/api/class/index.ts

```diff
@@ -37,12 +38,47 @@ export default async function handle(req: NextApiRequest, res: NextApiResponse):
             break;
         }
         case "POST": {
-            const classInput = req.body as ClassInput;
+            const classInput = req.body.classInput as ClassInput;
+            const classTranslationInput = req.body.classTranslationInput as ClassTranslationInput[];
+
+            //Create Stripe Key
+            const stripe = new Stripe(process.env.STRIPE_SECRET_KEY, { apiVersion: "2020-08-27" });
+
+            //If the price doesn't exist create a product and the associated price
+            if (!classInput.stripePriceId) {
+                const product = await stripe.products.create({
+                    name: classInput.name,
+                });
+                const stripePrice = await stripe.prices.create({
+                    unit_amount: parseInt(classInput.price),
+                    currency: "usd",
+                    product: product.id,
+                });
+
+                classInput.stripePriceId = stripePrice.id;
+            }
+            //Else if it exists, check if the price needs to be updated
+            else {
+                const price = await stripe.prices.retrieve(classInput.stripePriceId);
+                if (price.unit_amount !== parseInt(classInput.price)) {
+                    //Create a new price
+                    const stripePrice = await stripe.prices.create({
+                        unit_amount: parseInt(classInput.price),
+                        currency: "usd",
+                        product: price.product as string,
+                    });
+                    classInput.stripePriceId = stripePrice.id;
+                }
+            }
+
+            //Remove price as it was only needed to create the stripe price object
+            delete classInput.price;
+
```

Comment:
Please review this logic. Tested and works via the UI

Reply:
huh okay, 👀
Breaks our consistency a bit but this is fine. Remember that we only want to create products and prices if the class creation is successful.

Reply:
I can move this then...